added NAMEM as a new source

chunshek opened the pull request:
I have added National Agency for Meteorology and Environmental Monitoring (NAMEM) as a new official source for Mongolia. I have implemented the following:

- Current observation
- Hourly forecasts
- Daily forecasts
- Normals

Even though AQI is reported on the website, I did not implement air quality in this source. All but two of the air quality monitoring stations are located in Ulaanbaatar, so it's not useful for the rest of the country. Also, it only reports converted AQI numbers rather than raw concentrations, and I haven't find the conversion table for Mongolian AQI (yet).

papjul commented on app/src/src_nonfreenet/org/breezyweather/sources/namem/NamemResultConverter.kt:
```diff
@@ -0,0 +1,311 @@
+    context: Context,
+    weather: Int?,
+): String? {
+    // TODO: Check with a Mongolian speaker (???) for correct interpretation of the text
```
I asked for a proofreader on the Matrix channel. Will merge as is, if no one can help

chunshek commented:
> Unless the AQI is reported per pollutant (usually you get one single AQI for all pollutants), you can't convert back

The AQIs are reported per individual pollutant. If the conversion tables were published, I could theoretically implement Air Quality as well. But given the circumstances, I would say it's low priority.

papjul commented:
> All but two of the air quality monitoring stations are located in Ulaanbaatar, so it's not useful for the rest of the country.

Almost half of the population likes in Ulaanbaatar, so it's useful for most people, I would say?

Would this be possible to implement `isFeatureSupportedInMainForLocation()` in a way that makes `AIR_QUALITY` feature only available when the location coordinates are within 50 km of Ulanbaatar or 50 km of one of the two other stations? That would hardcode Ulanbaatar coordinates and two stations coordinates. It should be written instructions of how to update in case there are new stations and when it was last updated.

> I haven't find the conversion table for Mongolian AQI (yet)

If you can find the thresholds of AQI, I wouldn't be surprised it is based on China or US AQI.


We can have a look at it at a later time, no rush

chunshek commented:
Using a hyperlocal version of `isFeatureSupportedInMainForLocation()` is not a bad idea! I'll see if I could easily grab the coordinates of all those stations.

And… I found the [conversion table](http://agaar.mn/files/article/692/Агаарын%20чанарын%20индексээр%20агаарын%20чанарыг%20үнэлэх,%20мэдээлэх%20журам.pdf)! So I guess we might as well implement this before merging.

chunshek commented:
Updated with AQI conversion!

I didn't hyperlocalize `isFeatureSupportedInMainForLocation()`. Instead I grabbed the list of coordinates and located the nearest AQI station within a 50km radius, and added it as a location parameter. If there is no AQI station within that radius, the location parameter will remain empty, and the source will not poll AQI data for that location.

This way there is no need to hardcode the station coordinates and maintain the list.
